architecture-decision-records/0013-file-check-queues-and-lambdas.md

```diff
@@ -0,0 +1,185 @@
+# 13. Use queues and Lambdas to run backend file checks
+
+**Date:** 2021-01-27
+
+## Context
+
+When a user uploads a folder to a TDR consignment, TDR needs to run a set of
+checks on each of those files. For the MVP release, the checks are:
+
+- [Antivirus scan]
+- [SHA-256 checksum], which can be compared to the checksum calculated in the
+  user's browser
+- [File format ID], which can be used to highlight files that The National
+  Archives does not accept, like password-protected files, or files a user might
+  have uploaded by mistake, like executable files
+
```

Comment:
Is it worth saying that the file format ID is not only checking for invalid files and that we need the file format id for when it leaves TDR? Or are we not bothered because those uses are outside the scope of TDR? I don't mind either way.

Reply:
Yep - I've added that.